tests/std/tests/VSO_0000000_regex_use/test.cpp

```diff
@@ -582,6 +582,19 @@ void test_gh_993() {
     }
 }
 
+void test_gh_4995() {
+    // GH-4995: R"([\d-e])" should be rejected
+    g_regexTester.should_throw("[\\d-e]", error_range);
+    g_regexTester.should_throw("[e-\\d]", error_range);
+    g_regexTester.should_throw("[\\w-\\d]", error_range);
+    g_regexTester.should_throw("[[:digit:]-e]", error_range);
+    g_regexTester.should_throw("[e-[:digit:]]", error_range);
+    g_regexTester.should_throw("[[:alpha:]-[:digit:]]", error_range);
+    g_regexTester.should_throw("[[=a=]-e]", error_range, ECMAScript | regex::collate);
+    g_regexTester.should_throw("[e-[=a=]]", error_range, ECMAScript | regex::collate);
+    g_regexTester.should_throw("[[=a=]-[=b=]]", error_range, ECMAScript | regex::collate);
+}

```

Comment:
This is great test coverage of the error cases. As you're adjusting the control flow, let's also add positive test coverage of sets that should still be accepted.